### PR TITLE
Jennifer/gift exchange invited view

### LINKED ISSUE #46: [GiftExchange] GiftExchange Invited view
- [x] If a user (logged in or not) visits a GiftExchange they are not a member of. Show a modal asking if they want to join
- [x] Conditional logic for if a user is signed in or not
- [x] Conditional logic if a user is a member of the group or not
- [x] Display different buttons on different conditional values
Image

### PR DESCRIPTION
**Pull request by jennifertieu**
Closes #46 

- [x] If a user (logged in or not) visits a GiftExchange they are not a member of. Show a modal asking if they want to join
- [x] Conditional logic for if a user is signed in or not
- [x] Conditional logic if a user is a member of the group or not
- [x] Display different buttons on different conditional values

Please note:
- For the `/gift-exchange` API get request, I removed the unauthorized user check to allow the page to be viewed by a user who is not logged in. 
- For the middleware, if the user is on the gift exchange group page the user is NOT going to be redirected to login page if they are not logged in for the same reason as above

Image

### CODE REVIEW

File: app/gift-exchanges/[id]/page.tsx [hunk 1]
```diff
@@ -131,7 +160,15 @@ export default function GiftExchangePage() {
...
   return (
-    <main className="min-h-screen">
+    <main className="h-screen">
+      {!isUserAMember && (
+        <WarningModal
+          giftExchangeData={giftExchangeData}
+          session={session}
+          members={giftExchangeMembers}
+          updateGiftExchangeMembers={updateGiftExchangeMembers}
+        />
```
Comment (thomasnguyen) [inline app/gift-exchanges/[id]/page.tsx:170]: I believe we need to also only show this if they status of the gift exchange is pending. Do we cover that here?

File: app/gift-exchanges/[id]/WarningModal.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+              <div className="text-sm">${giftExchangeData.budget}</div>
+            </div>
+            <div>
+              <div className="text-sm font-bold">Members</div>
+              <div className="text-sm">{members?.length || 0}</div>
+            </div>
+          </div>
+          {session ? (
+            <Button className="w-full" onClick={joinExchange}>
+              Join <LogIn />
```
Comment (thomasnguyen) [inline app/gift-exchanges/[id]/WarningModal.tsx:69]: We probably dont need the login button here if they are already logged in right?

File: app/gift-exchanges/[id]/WarningModal.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            </Button>
+          ) : (
+            <Button className="w-full flex items-center gap-2" onClick={signIn}>
+              Sign in with Google to Join <LogIn />
+            </Button>
```
Comment (thomasnguyen) [inline app/gift-exchanges/[id]/WarningModal.tsx:74]: Oh nice this is pretty cool! Appreciate you setting up the redirects and stuffs

File: app/gift-exchanges/[id]/page.tsx (hunk 1 above)
Comment (jennifertieu) [reply app/gift-exchanges/[id]/page.tsx:170]: Ah good point, they shouldn't be able to join if the gift exchange is done. Good catch! I'll update it, thank you @thomasnguyen

File: app/gift-exchanges/[id]/WarningModal.tsx (hunk 2 above)
Comment (jennifertieu) [reply app/gift-exchanges/[id]/WarningModal.tsx:69]: True, I'll remove the LogIn icon

File: app/gift-exchanges/[id]/WarningModal.tsx (hunk 3 above)
Comment (jennifertieu) [reply app/gift-exchanges/[id]/WarningModal.tsx:74]: No problem! I did everything as fast as possible so hopefully, that should work. I tested it a few times so we should be okay

**Pull request merged**